static/js/services/enketo-translation.js

```diff
@@ -327,16 +327,13 @@ angular.module('inboxServices').service('EnketoTranslation', [
       return res;
     };
 
-    self.bindJsonToXml = function(elem, data) {
-      // Pairs get sorted so that values are bound before objects:
-      // - data is bound to any decendents of the given elem
-      // - least specific (in terms of path) get bound first
-      // - more specific get to bind *over* potentially too unspecific values
-      _.sortBy(_.pairs(data), function(pair) {
-        return _.isObject(pair[1]);
-      }).forEach(function(pair) {
-        var current = elem.find(pair[0]);
+    // NB: childMatcher intentionally does not recurse.
+    //     It exists to allow the initial layer of binding to be flexible.
+    self.bindJsonToXml = function(elem, data, childMatcher) {
+      _.pairs(data).forEach(function(pair) {
+        var current = childMatcher ? elem.find(childMatcher([pair[0]])) : elem.children(pair[0]);
         var value = pair[1];
+
         if (_.isObject(value)) {
           if(current.children().length) {
             self.bindJsonToXml(current, value);
```

Comment:
@garethbowen Yeah that's a great question.

On the one hand, that's what the old code does. so we've been living with it for some time now. I also am not entirely sure what we'd do if the `find` came up with more than one option: error out? Pick the first one? Attempt to somehow pick the most specific one? Just log a warn and call it a day?

For now I'll just log it, but if you think any of the other options listed (or one I didn't list) is better we should definitely change it, I just don't really have the confidence to do anything more shocking.

I think / hope that this is mostly dealt with by the flexibility of find only occurring on the first bind level, which should make it reasonably easy to avoid.